smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/model/SymbolBuilder.kt

```diff
@@ -113,9 +113,10 @@ fun Symbol.isInternalSPI(): Boolean {
     return this.getProperty("isInternalSPI").orElse(false) as Boolean
 }
 
-fun Symbol.toInternalSPI(kind: SwiftKind): Symbol {
+fun Symbol.toInternalSPI(kind: SwiftDeclaration, spiName: String): Symbol {
     return this.toBuilder()
         .putProperty("isInternalSPI", true)
+        .putProperty("spiName", spiName)
         .putProperty("kind", kind.kind)
```

Comment:
do we still need this `putProperty` for kind?

Reply:
I think doing so would be a good way to specify exactly what's being imported with SPI. 
Allows us to avoid having 
```
@_spi(InternalA) import AWSClientRuntime
@_spi(InternalB) import AWSClientRuntime
```
And instead have
```
@_spi(InternalA) import struct AWSClientRuntime.InternalA
@_spi(InternalB) import enum AWSClientRuntime.InternalB
```
For use case where there're multiple things under same spi tag, that can be revisited down the line if it pops up.